R/checks.R

```diff
@@ -2358,26 +2358,14 @@ checkBadFiles <- function(package_dir){
         return(invisible())
     }
     licenses <- read.dcf(ldb_file)
-    sss <- licenses[, "SSS"]
-    abbrev <- licenses[, "Abbrev"]
-    abbrev[!is.na(sss)] <- sss[!is.na(sss)]
-    restrict <- licenses[, "Restricts_use"]
-    idx <- (restrict == "yes") & !is.na(restrict) & !is.na(abbrev)
-
-    ## PCAN/DESCRIPTION:License: CC BY-NC-ND 4.0
-    ## QUBIC/DESCRIPTION:License: CC BY-NC-ND 4.0 + file LICENSE
-    test0 <- any(vapply(abbrev[idx], grepl, logical(1), license, fixed = TRUE))
-    test1 <-
-        !any(vapply(
-             abbrev[!is.na(abbrev)], grepl, logical(1), license, fixed = TRUE
-         ))
-    if (test0) {
+    result <- tools:::analyze_licenses(license, licenses)
```

Comment:
this function isn't exported. Add a unit test that ensures that it still exists, and behaves as expected in at least some cases.